qrenderdoc/Windows/TextureViewer.cpp

```diff
@@ -412,14 +425,48 @@ class TextureListItemModel : public QAbstractItemModel
     return QAbstractItemModel::flags(index);
   }
 
+  QVariant headerData(int section, Qt::Orientation orientation, int role) const override
+  {
+    if(orientation == Qt::Horizontal && role == Qt::DisplayRole)
+    {
+      switch(section)
+      {
+        case Column_TexName: return lit("Texture Name");
+        case Column_TexWidth: return lit("Width");
+        case Column_TexHeight: return lit("Height");
+        case Column_TexDepth: return lit("Depth");
+        case Column_MipsCount: return lit("Mips Count");
+        case Column_TexFormat: return lit("Format");
+        default: break;
+      }
+    }
+
+    return QVariant();
+  }
+
   QVariant data(const QModelIndex &index, int role = Qt::DisplayRole) const override
   {
     if(index.isValid())
     {
       if(role == Qt::DisplayRole)
       {
-        if(index.row() >= 0 && index.row() < texs.count())
-          return m_Ctx.GetResourceName(texs[index.row()].resourceId);
+        int row = index.row();
+        int col = index.column();
+
+        if(col >= 0 && col < columnCount() && row < rowCount())
+        {
+          QString name = m_Ctx.GetResourceName(texs[index.row()].resourceId);
+          switch(col)
+          {
+            case Column_TexName: return m_Ctx.GetResourceName(texs[index.row()].resourceId);
+            case Column_TexWidth: return QString::number(texs[index.row()].width);
+            case Column_TexHeight: return QString::number(texs[index.row()].height);
+            case Column_TexDepth: return QString::number(texs[index.row()].depth);
```

Comment:
Would it be useful here to return the depth for 3D textures, and the array size for non-3D textures? Since they're disjoint (you can't have 3D arrays).